commit 2040a07526dca7af955f293902627d54b0e188ee
Author: Tomas Pizarro <tompizmor@users.noreply.github.com>
Date:   Wed Apr 4 13:10:07 2018 +0200

     [stable/redis] Refactor image properties in values.yaml (#4674)
    
    * [stable/redis] Refactor image properties in values.yaml
    
    * Use rolling tag and refactor pullSecrets
    
    * Fix lint issue
    
    * Bump chart version
    
    * Fix deployment image

diff --git a/stable/redis/Chart.yaml b/stable/redis/Chart.yaml
index 6ffdd2383..265258cab 100644
--- a/stable/redis/Chart.yaml
+++ b/stable/redis/Chart.yaml
@@ -1,5 +1,5 @@
 name: redis
-version: 1.1.24
+version: 1.2.0
 appVersion: 4.0.9
 description: Open source, advanced key-value store. It is often referred to as a data structure server since keys can contain strings, hashes, lists, sets and sorted sets.
 keywords:
diff --git a/stable/redis/README.md b/stable/redis/README.md
index 76ddd9585..3b21506c6 100644
--- a/stable/redis/README.md
+++ b/stable/redis/README.md
@@ -43,36 +43,39 @@ The command removes all the Kubernetes components associated with the chart and
 
 The following table lists the configurable parameters of the Redis chart and their default values.
 
-|           Parameter           |                Description                        |           Default            |
-|-------------------------------|-------------------------------------------------- |------------------------------|
-| `image`                       | Redis image                                       | `bitnami/redis:{VERSION}`    |
-| `imagePullPolicy`             | Image pull policy                                 | `IfNotPresent`               |
-| `serviceType`                 | Kubernetes Service type                           | `ClusterIP`                  |
-| `usePassword`                 | Use password                                      | `true`                       |
-| `redisPassword`               | Redis password                                    | Randomly generated           |
-| `redisDisableCommands`        | Comma-separated list of Redis commands to disable | `FLUSHDB,FLUSHALL`           |
-| `args`                        | Redis command-line args                           | []                           |
-| `redisExtraFlags`             | Redis additional command line flags               | []                           |
-| `persistence.enabled`         | Use a PVC to persist data                         | `true`                       |
-| `persistence.path`            | Path to mount the volume at, to use other images  | `/bitnami`                   |
-| `persistence.subPath`         | Subdirectory of the volume to mount at            | `""`                         |
-| `persistence.existingClaim`   | Use an existing PVC to persist data               | `nil`                        |
-| `persistence.storageClass`    | Storage class of backing PVC                      | `generic`                    |
-| `persistence.accessMode`      | Use volume as ReadOnly or ReadWrite               | `ReadWriteOnce`              |
-| `persistence.size`            | Size of data volume                               | `8Gi`                        |
-| `resources`                   | CPU/Memory resource requests/limits               | Memory: `256Mi`, CPU: `100m` |
-| `metrics.enabled`             | Start a side-car prometheus exporter              | `false`                      |
-| `metrics.image`               | Exporter image                                    | `oliver006/redis_exporter`   |
-| `metrics.imageTag`            | Exporter image                                    | `v0.11`                      |
-| `metrics.imagePullPolicy`     | Exporter image pull policy                        | `IfNotPresent`               |
-| `metrics.resources`           | Exporter resource requests/limit                  | Memory: `256Mi`, CPU: `100m` |
-| `nodeSelector`                | Node labels for pod assignment                    | {}                           |
-| `tolerations`                 | Toleration labels for pod assignment              | []                           |
-| `networkPolicy.enabled`       | Enable NetworkPolicy                              | `false`                      |
-| `networkPolicy.allowExternal` | Don't require client label for connections        | `true`                       |
-| `service.annotations`         | annotations for redis service                     | {}                           |
-| `service.loadBalancerIP`      | loadBalancerIP if service type is `LoadBalancer`  | ``                           |
-| `securityContext.enabled`     | Enable security context                           | `true`                       |
+|           Parameter           |                Description                        |                      Default                            |
+|-------------------------------|-------------------------------------------------- |-------------------------------------------------------- |
+| `image.registry`              | Redis image registry                              | `docker.io`                                             |
+| `image.repository`            | Redis image name                                  | `bitnami/redis`                                         |
+| `image.tag`                   | Redis image tag                                   | `{VERSION}`                                             |
+| `image.pullPolicy`            | Image pull policy                                 | `Always` if `imageTag` is `latest`, else `IfNotPresent` |
+| `image.pullSecrets`           | Specify image pull secrets                        | `nil`                                                   |
+| `serviceType`                 | Kubernetes Service type                           | `ClusterIP`                                             |
+| `usePassword`                 | Use password                                      | `true`                                                  |
+| `redisPassword`               | Redis password                                    | Randomly generated                                      |
+| `redisDisableCommands`        | Comma-separated list of Redis commands to disable | `FLUSHDB,FLUSHALL`                                      |
+| `args`                        | Redis command-line args                           | []                                                      |
+| `redisExtraFlags`             | Redis additional command line flags               | []                                                      |
+| `persistence.enabled`         | Use a PVC to persist data                         | `true`                                                  |
+| `persistence.path`            | Path to mount the volume at, to use other images  | `/bitnami`                                              |
+| `persistence.subPath`         | Subdirectory of the volume to mount at            | `""`                                                    |
+| `persistence.existingClaim`   | Use an existing PVC to persist data               | `nil`                                                   |
+| `persistence.storageClass`    | Storage class of backing PVC                      | `generic`                                               |
+| `persistence.accessMode`      | Use volume as ReadOnly or ReadWrite               | `ReadWriteOnce`                                         |
+| `persistence.size`            | Size of data volume                               | `8Gi`                                                   |
+| `resources`                   | CPU/Memory resource requests/limits               | Memory: `256Mi`, CPU: `100m`                            |
+| `metrics.enabled`             | Start a side-car prometheus exporter              | `false`                                                 |
+| `metrics.image`               | Exporter image                                    | `oliver006/redis_exporter`                              |
+| `metrics.imageTag`            | Exporter image                                    | `v0.11`                                                 |
+| `metrics.imagePullPolicy`     | Exporter image pull policy                        | `IfNotPresent`                                          |
+| `metrics.resources`           | Exporter resource requests/limit                  | Memory: `256Mi`, CPU: `100m`                            |
+| `nodeSelector`                | Node labels for pod assignment                    | {}                                                      |
+| `tolerations`                 | Toleration labels for pod assignment              | []                                                      |
+| `networkPolicy.enabled`       | Enable NetworkPolicy                              | `false`                                                 |
+| `networkPolicy.allowExternal` | Don't require client label for connections        | `true`                                                  |
+| `service.annotations`         | annotations for redis service                     | {}                                                      |
+| `service.loadBalancerIP`      | loadBalancerIP if service type is `LoadBalancer`  | ``                                                      |
+| `securityContext.enabled`     | Enable security context                           | `true`                                                  |
 
 The above parameters map to the env variables defined in [bitnami/redis](http://github.com/bitnami/bitnami-docker-redis). For more information please refer to the [bitnami/redis](http://github.com/bitnami/bitnami-docker-redis) image documentation.
 
diff --git a/stable/redis/templates/deployment.yaml b/stable/redis/templates/deployment.yaml
index e24c538f9..641bf543d 100644
--- a/stable/redis/templates/deployment.yaml
+++ b/stable/redis/templates/deployment.yaml
@@ -32,11 +32,17 @@ spec:
       {{- if .Values.tolerations }}
       tolerations:
 {{ toYaml .Values.tolerations | indent 8 }}
+      {{- end }}
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
       {{- end }}
       containers:
       - name: {{ template "redis.fullname" . }}
-        image: "{{ .Values.image }}"
-        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         {{- if .Values.args }}
         args:
 {{ toYaml .Values.args | indent 10 }}
diff --git a/stable/redis/values.yaml b/stable/redis/values.yaml
index c9a4ffbb4..f1648d71f 100644
--- a/stable/redis/values.yaml
+++ b/stable/redis/values.yaml
@@ -1,12 +1,21 @@
 ## Bitnami Redis image version
 ## ref: https://hub.docker.com/r/bitnami/redis/tags/
 ##
-image: bitnami/redis:4.0.9-r0
-
-## Specify a imagePullPolicy
-## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
-##
-imagePullPolicy: IfNotPresent
+image:
+  registry: docker.io
+  repository: bitnami/redis
+  tag: 4.0.9
+  ## Specify a imagePullPolicy
+  ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
+  ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
+  ##
+  pullPolicy: IfNotPresent
+  ## Optionally specify an array of imagePullSecrets.
+  ## Secrets must be manually created in the namespace.
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+  ##
+  # pullSecrets:
+  #   - myRegistrKeySecretName
 
 ## Kubernetes service type
 serviceType: ClusterIP
